core/common/utils/StreamUtilities.hpp

```diff
@@ -32,10 +32,26 @@
 #include <ostream>
 #include <set>
 #include <vector>
+#include <tuple>
 
 
 namespace netmeld::core::utils {
 
+  template<size_t Idx = 0, typename... Types>
+  std::ostream& operator<<(std::ostream& os, const std::tuple<Types...>& tup)
+  {
+    if constexpr(Idx == 0) {
+      os << "[";
+    }
+    os << std::get<Idx>(tup);
+    if constexpr(Idx + 1 < sizeof...(Types)) {
+      os << " -- ";
```

Comment:
Not sure what the intent was here.  If it was to provide a separator between the tuple values, then please just use a comma then space (`, `) instead of a space, double dash, and space(` -- `).